### ISSUE TITLE
[Feature Request]: use separate repo as source of markdown files to generate doc pages

### ISSUE THREAD

**Issue by mahula** [vuepress-theme-hope/vuepress-theme-hope#3591]
### Checklist

- [X] I searched existing issues and no one else requests similar feature.
- [X] I think that 25%+ users are positive towards this feature.

### Describe the feature

Separating code and deployment infratstructure from the documentation configuration, it would be nice to have all the Vuepress configuration in a separate repo.
Can it be possible, to set in a mere documentation repo `docsRepo` to a separate code repo's URL to get all Markdown files from there?
For example, to fully independently configure all Vuepress and Hope-Theme related things to create a fine documentation website related to a code-based repo as [Ocelot-Social](https://github.com/Ocelot-Social-Community/Ocelot-Social), it would be nice to not have to copy allMarkdown files from there to the documentation website repo.

**Comment by Mister-Hope** [comment]
Actually I'm not getting your point.

VuePress does not have the ability  to generate site from source code.

In another world,  You should generate all your Markdown files before starting to build a site with vuepress,  Then you can just add an extra script to copy the generated Mark down together with the site to another branch of repo and deploy it
